src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java

```diff
@@ -181,11 +194,47 @@ private void checkParagraphTag(DetailNode tag) {
         else if (newLine == null || tag.getLineNumber() - newLine.getLineNumber() != 1) {
             log(tag.getLineNumber(), MSG_LINE_BEFORE);
         }
+
+        checkIfFollowedByBlockTag(tag);
+
         if (allowNewlineParagraph && isImmediatelyFollowedByText(tag)) {
             log(tag.getLineNumber(), MSG_MISPLACED_TAG);
         }
     }
 
+    /**
+     * Determines whether or not the paragraph tag is followed by block tag.
+     *
+     * @param tag html tag.
+     */
+    private void checkIfFollowedByBlockTag(DetailNode tag) {
+        DetailNode htmlElement = JavadocUtil.getNextSibling(tag);
+        while (htmlElement != null && htmlElement.getType() != JavadocTokenTypes.HTML_ELEMENT) {
+            if (htmlElement.getType() == JavadocTokenTypes.TEXT
+                    && !CommonUtil.isBlank(htmlElement.getText())) {
+                htmlElement = null;
+                break;
+            }
+            htmlElement = JavadocUtil.getNextSibling(htmlElement);
+        }
+
+        if (htmlElement != null) {
+            final DetailNode htmlTag = JavadocUtil.getFirstChild(htmlElement);
+            final Set<String> blockTags = new HashSet<>(Arrays.asList("address", "blockquote",
+                    "div", "dl", "h1", "h2", "h3", "h4", "h5", "h6", "hr", "ol",
+                    "p", "pre", "table", "ul"));
```

Comment:
make it as static final class field to be create ones.

Reply:
done